MaryTTS test test_service_say fails intermittently because it is not mocking the request

## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->

`2020-08-30T15:47:25.3527794Z FAILED tests/components/marytts/test_tts.py::TestTTSMaryTTSPlatform::test_service_say`
```
```
2020-08-30T15:47:25.2898536Z =================================== FAILURES ===================================
2020-08-30T15:47:25.2898714Z ___________________ TestTTSMaryTTSPlatform.test_service_say ____________________
2020-08-30T15:47:25.2901014Z [gw1] linux -- Python 3.8.2 /__w/core/core/venv/bin/python
2020-08-30T15:47:25.2901146Z 
2020-08-30T15:47:25.2901315Z self = <tests.components.marytts.test_tts.TestTTSMaryTTSPlatform object at 0x7f92e266c6d0>
2020-08-30T15:47:25.2901430Z 
2020-08-30T15:47:25.2901579Z     def test_service_say(self):
2020-08-30T15:47:25.2901721Z         """Test service call say."""
2020-08-30T15:47:25.2901893Z         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
2020-08-30T15:47:25.2902055Z     
2020-08-30T15:47:25.2902201Z         conn = Mock()
2020-08-30T15:47:25.2902352Z         response = Mock()
2020-08-30T15:47:25.2902513Z         conn.getresponse.return_value = response
2020-08-30T15:47:25.2902675Z         response.status = 200
2020-08-30T15:47:25.2902836Z         response.read.return_value = b"audio"
2020-08-30T15:47:25.2902992Z     
2020-08-30T15:47:25.2903149Z         config = {tts.DOMAIN: {"platform": "marytts"}}
2020-08-30T15:47:25.2903301Z     
2020-08-30T15:47:25.2903452Z         with assert_setup_component(1, tts.DOMAIN):
2020-08-30T15:47:25.2903628Z             setup_component(self.hass, tts.DOMAIN, config)
2020-08-30T15:47:25.2903782Z     
2020-08-30T15:47:25.2903954Z         with patch("http.client.HTTPConnection", return_value=conn):
2020-08-30T15:47:25.2904132Z             self.hass.services.call(
2020-08-30T15:47:25.2904293Z                 tts.DOMAIN,
2020-08-30T15:47:25.2904456Z                 "marytts_say",
2020-08-30T15:47:25.2904613Z                 {
2020-08-30T15:47:25.2904780Z                     "entity_id": "media_player.something",
2020-08-30T15:47:25.2904957Z                     tts.ATTR_MESSAGE: "HomeAssistant",
2020-08-30T15:47:25.2905121Z                 },
2020-08-30T15:47:25.2905272Z             )
2020-08-30T15:47:25.2905428Z         self.hass.block_till_done()
2020-08-30T15:47:25.2905733Z     
2020-08-30T15:47:25.2905862Z >       assert len(calls) == 1
2020-08-30T15:47:25.2906008Z E       assert 0 == 1
2020-08-30T15:47:25.2906152Z E        +  where 0 = len([])
2020-08-30T15:47:25.2906247Z 
2020-08-30T15:47:25.2906401Z tests/components/marytts/test_tts.py:87: AssertionError
```
2020-08-30T15:47:25.2928684Z DEBUG    homeassistant.core:core.py:639 Bus:Handling <Event component_loaded[L]: component=tts>
2020-08-30T15:47:25.2928899Z DEBUG    homeassistant.core:core.py:639 Bus:Handling <Event call_service[L]: domain=tts, service=marytts_say, service_data=entity_id=media_player.something, message=HomeAssistant>
2020-08-30T15:47:25.2929491Z ERROR    homeassistant.core:core.py:1329 Error executing service: <ServiceCall tts.marytts_say (c:f45960d5ead711ea947fb50da2ed6630): entity_id=['media_player.something'], message=HomeAssistant>
2020-08-30T15:47:25.2929682Z Traceback (most recent call last):
2020-08-30T15:47:25.2929965Z   File "/__w/core/core/homeassistant/core.py", line 1319, in catch_exceptions
2020-08-30T15:47:25.2930150Z     await coro_or_task
2020-08-30T15:47:25.2930389Z   File "/__w/core/core/homeassistant/core.py", line 1338, in _execute_service
2020-08-30T15:47:25.2930602Z     await handler.func(service_call)
2020-08-30T15:47:25.2930857Z   File "/__w/core/core/homeassistant/components/tts/__init__.py", line 167, in async_say_handle
2020-08-30T15:47:25.2931036Z     url = await tts.async_get_url(
2020-08-30T15:47:25.2931340Z   File "/__w/core/core/homeassistant/components/tts/__init__.py", line 340, in async_get_url
2020-08-30T15:47:25.2931645Z     filename = await self.async_get_tts_audio(
2020-08-30T15:47:25.2931935Z   File "/__w/core/core/homeassistant/components/tts/__init__.py", line 352, in async_get_tts_audio
2020-08-30T15:47:25.2932136Z     extension, data = await provider.async_get_tts_audio(message, language, options)
2020-08-30T15:47:25.2932400Z   File "/__w/core/core/homeassistant/components/tts/__init__.py", line 515, in async_get_tts_audio
2020-08-30T15:47:25.2932586Z     return await self.hass.async_add_executor_job(
2020-08-30T15:47:25.2932809Z   File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2020-08-30T15:47:25.2932991Z     result = self.fn(*self.args, **self.kwargs)
2020-08-30T15:47:25.2933281Z   File "/__w/core/core/homeassistant/components/marytts/tts.py", line 87, in get_tts_audio
2020-08-30T15:47:25.2933466Z     data = self._mary.speak(message, effects)
2020-08-30T15:47:25.2934079Z   File "/__w/core/core/venv/lib/python3.8/site-packages/speak2mary/__init__.py", line 48, in speak
2020-08-30T15:47:25.2934281Z     conn.request("POST", "/process", urlencode(raw_params))
2020-08-30T15:47:25.2934513Z   File "/usr/local/lib/python3.8/http/client.py", line 1230, in request
2020-08-30T15:47:25.2934697Z     self._send_request(method, url, body, headers, encode_chunked)
2020-08-30T15:47:25.2934926Z   File "/usr/local/lib/python3.8/http/client.py", line 1276, in _send_request
2020-08-30T15:47:25.2935114Z     self.endheaders(body, encode_chunked=encode_chunked)
2020-08-30T15:47:25.2935360Z   File "/usr/local/lib/python3.8/http/client.py", line 1225, in endheaders
2020-08-30T15:47:25.2935547Z     self._send_output(message_body, encode_chunked=encode_chunked)
2020-08-30T15:47:25.2935768Z   File "/usr/local/lib/python3.8/http/client.py", line 1004, in _send_output
2020-08-30T15:47:25.2938148Z     self.send(msg)
2020-08-30T15:47:25.2938383Z   File "/usr/local/lib/python3.8/http/client.py", line 944, in send
2020-08-30T15:47:25.2938553Z     self.connect()
2020-08-30T15:47:25.2938765Z   File "/usr/local/lib/python3.8/http/client.py", line 915, in connect
2020-08-30T15:47:25.2938945Z     self.sock = self._create_connection(
2020-08-30T15:47:25.2939156Z   File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
2020-08-30T15:47:25.2939336Z     raise err
2020-08-30T15:47:25.2939539Z   File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
2020-08-30T15:47:25.2939719Z     sock.connect(sa)
2020-08-30T15:47:25.2939872Z OSError: [Errno 99] Cannot assign requested address
```

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -65,7 +65,10 @@ class TestTTSMaryTTSPlatform:
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch("homeassistant.components.marytts.tts.MaryTTS.speak") as mock_speak:
+        with patch(
+            "homeassistant.components.marytts.tts.MaryTTS.speak",
+            return_value="audio",
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",
@@ -93,7 +96,10 @@ class TestTTSMaryTTSPlatform:
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch("homeassistant.components.marytts.tts.MaryTTS.speak") as mock_speak:
+        with patch(
+            "homeassistant.components.marytts.tts.MaryTTS.speak",
+            return_value="audio",
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -10,10 +10,9 @@ from homeassistant.components.media_player.const import (
 )
 import homeassistant.components.tts as tts
 from homeassistant.config import async_process_ha_core_config
-from homeassistant.const import HTTP_INTERNAL_SERVER_ERROR
 from homeassistant.setup import setup_component
 
-from tests.async_mock import Mock, patch
+from tests.async_mock import patch
 from tests.common import assert_setup_component, get_test_home_assistant, mock_service
 
 
@@ -81,7 +80,8 @@ class TestTTSMaryTTSPlatform:
 
         assert len(calls) == 1
         assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
-        conn.request.assert_called_with("POST", "/process", urlencode(self.params))
+
+        mock_speak.assert_called_with("HomeAssistant", {})
 
     def test_service_say_with_effect(self):
         """Test service call say with effects."""
@@ -110,10 +110,7 @@ class TestTTSMaryTTSPlatform:
         assert len(calls) == 1
         assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
 
-        self.params.update(
-            {"effect_Volume_selected": "on", "effect_Volume_parameters": "amount:2.0;"}
-        )
-        conn.request.assert_called_with("POST", "/process", urlencode(self.params))
+        mock_speak.assert_called_with("HomeAssistant", {"Volume": "amount:2.0;"})
 
     def test_service_say_http_error(self):
         """Test service call say."""
@@ -126,6 +123,7 @@ class TestTTSMaryTTSPlatform:
 
         with patch(
             "homeassistant.components.marytts.tts.MaryTTS.speak",
+            side_effect=Exception(),
         ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
@@ -138,4 +136,3 @@ class TestTTSMaryTTSPlatform:
         self.hass.block_till_done()
 
         assert len(calls) == 0
-        conn.request.assert_called_with("POST", "/process", urlencode(self.params))

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -2,7 +2,6 @@
 import asyncio
 import os
 import shutil
-from urllib.parse import urlencode
 
 from homeassistant.components.media_player.const import (
     ATTR_MEDIA_CONTENT_ID,
@@ -62,18 +61,14 @@ class TestTTSMaryTTSPlatform:
         """Test service call say."""
         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        conn = Mock()
-        response = Mock()
-        conn.getresponse.return_value = response
-        response.status = 200
-        response.read.return_value = b"audio"
-
         config = {tts.DOMAIN: {"platform": "marytts"}}
 
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch("http.client.HTTPConnection", return_value=conn):
+        with patch(
+            "homeassistant.components.marytts.tts.MaryTTS.speak"
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",
@@ -92,12 +87,6 @@ class TestTTSMaryTTSPlatform:
         """Test service call say with effects."""
         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        conn = Mock()
-        response = Mock()
-        conn.getresponse.return_value = response
-        response.status = 200
-        response.read.return_value = b"audio"
-
         config = {
             tts.DOMAIN: {"platform": "marytts", "effect": {"Volume": "amount:2.0;"}}
         }
@@ -105,7 +94,9 @@ class TestTTSMaryTTSPlatform:
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch("http.client.HTTPConnection", return_value=conn):
+        with patch(
+            "homeassistant.components.marytts.tts.MaryTTS.speak"
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",
@@ -128,19 +119,14 @@ class TestTTSMaryTTSPlatform:
         """Test service call say."""
         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        conn = Mock()
-        response = Mock()
-        conn.getresponse.return_value = response
-        response.status = HTTP_INTERNAL_SERVER_ERROR
-        response.reason = "test"
-        response.readline.return_value = "content"
-
         config = {tts.DOMAIN: {"platform": "marytts"}}
 
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch("http.client.HTTPConnection", return_value=conn):
+        with patch(
+            "homeassistant.components.marytts.tts.MaryTTS.speak",
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -67,7 +67,7 @@ class TestTTSMaryTTSPlatform:
 
         with patch(
             "homeassistant.components.marytts.tts.MaryTTS.speak",
-            return_value="audio",
+            return_value=b"audio",
         ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
@@ -98,7 +98,7 @@ class TestTTSMaryTTSPlatform:
 
         with patch(
             "homeassistant.components.marytts.tts.MaryTTS.speak",
-            return_value="audio",
+            return_value=b"audio",
         ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -65,9 +65,7 @@ class TestTTSMaryTTSPlatform:
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch(
-            "homeassistant.components.marytts.tts.MaryTTS.speak"
-        ) as mock_speak:
+        with patch("homeassistant.components.marytts.tts.MaryTTS.speak") as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",
@@ -95,9 +93,7 @@ class TestTTSMaryTTSPlatform:
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        with patch(
-            "homeassistant.components.marytts.tts.MaryTTS.speak"
-        ) as mock_speak:
+        with patch("homeassistant.components.marytts.tts.MaryTTS.speak") as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -126,7 +126,7 @@ class TestTTSMaryTTSPlatform:
         with patch(
             "homeassistant.components.marytts.tts.MaryTTS.speak",
             side_effect=Exception(),
-        )as mock_speak:
+        ) as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",

--- a/tests/components/marytts/test_tts.py
+++ b/tests/components/marytts/test_tts.py
@@ -81,6 +81,7 @@ class TestTTSMaryTTSPlatform:
         assert len(calls) == 1
         assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
 
+        mock_speak.assert_called_once()
         mock_speak.assert_called_with("HomeAssistant", {})
 
     def test_service_say_with_effect(self):
@@ -110,6 +111,7 @@ class TestTTSMaryTTSPlatform:
         assert len(calls) == 1
         assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
 
+        mock_speak.assert_called_once()
         mock_speak.assert_called_with("HomeAssistant", {"Volume": "amount:2.0;"})
 
     def test_service_say_http_error(self):
@@ -124,7 +126,7 @@ class TestTTSMaryTTSPlatform:
         with patch(
             "homeassistant.components.marytts.tts.MaryTTS.speak",
             side_effect=Exception(),
-        ) as mock_speak:
+        )as mock_speak:
             self.hass.services.call(
                 tts.DOMAIN,
                 "marytts_say",
@@ -136,3 +138,5 @@ class TestTTSMaryTTSPlatform:
         self.hass.block_till_done()
 
         assert len(calls) == 0
+
+        mock_speak.assert_called_once()

